src/components/MoleculeRepresentation/MoleculeRepresentation.tsx

```diff
@@ -112,27 +116,43 @@ export const MoleculeRepresentation: React.FC<MoleculeRepresentationProps> = mem
       alignmentDetails,
     ]);
 
-    if (showLoadingSpinner && !svgContent) return <Spinner width={width} height={height} />;
+    const handleOnClick = useCallback(
+      (e: React.MouseEvent) => {
+        const clickedId = (e.target as SVGRectElement).id;
+        if (onAtomClick && clickedId) {
+          e.preventDefault();
+          e.stopPropagation();
+          const atomIdx = getAtomIdxFromClickableId(clickedId);
+          onAtomClick(atomIdx);
+        }
+      },
+      [onAtomClick],
+    );
+
+    if (!svgContent) {
+      if (showLoadingSpinner) return <Spinner width={width} height={height} />;
+      else return null;
+    }
+
+    const svgElement = createSvgElement(svgContent, {
```

Comment:
why do we need `createSvgElement` and the new library `react-from-dom`?

Reply:
We need to convert the svg string to a react/dom element so we can encapsulate it inside <g> tag here
https://github.com/iktos/molecule-representation/pull/28/files#diff-4287dc2e5fb94763a0a8d6ff53de09a6957558d09da6f82156352293bd4dda7bR99
It's needed for the dragging feature when zooming.

Reply:
why not just use `dangerouslySetInnerHTML` to do so?